lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -139,95 +150,275 @@
           }
         }
 '''
+import json
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.basic import AnsibleModule, get_all_subclasses
+from ansible.module_utils.six import with_metaclass
+
+
+class TestFailed(Exception):
+    pass
+
+
+class PkgMgr(with_metaclass(ABCMeta, object)):
+
+    def __init__(self):
+
+        if not self.test():
+            raise TestFailed("Test failed to pass")
```

Comment:
This looks like there's some testing in place, while it's just a conditional prerequisites check. It's much more friendly if you put some clue about what's happening there:
```suggestion
            raise PackageManagerMissing("%s is not installed or unavailable" % (self.CLI, ))
```